src/utils/use-is-in-vegetation-period.tsx

```diff
@@ -0,0 +1,9 @@
+import { isWithinInterval } from "date-fns";
+
+const currentDate = new Date();
+const currentYear = currentDate.getFullYear();
+
+export const useIsInVegetationPeriod = isWithinInterval(currentDate, {
```

Comment:
Two points here:

- the `use...` is usually a convention for hooks (functions) that are called in a react component. If you want to keep it as const, i'd suggest to directly name it: `isInVegetationPeriod` . That way you also don't need to import it and assign it to another constant in the components you need it. 
- By declaring it as a constant the value will be calculated at the moment of page load. This can lead to edge cases, e.g. open the page in october, leave it open until november, and you'll still see everything as in vegetation period. I'd suggest to declare it as function and calculate the value every time you need it, (and inside a component, so new renderings will take it into account). If you chose to use a function, you can keep the `use` prefix 👍 

let me know what you think :)

Reply:
Thanks for the clarification! Could you have another look? I chose to declare it as a function to avoid edge cases.